catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java

```diff
@@ -384,7 +384,7 @@ public void refresh(Map<String, Object> configuration) throws SecurityServiceExc
 
         String idProp = (String) configuration.get(ID_PROPERTY);
         if (StringUtils.isNotBlank(idProp)) {
-            cswSourceConfiguration.setId(idProp);
+            setId(idProp);
```

Comment:
The method setId() sets the cswSourceConfiguration ID and sets the ID in super.  The subsequent cswSourceConfiguration.getId() calls should return the right value.